beetle-core/src/main/java/com/xing/beetle/dedup/spi/KeyValueStoreBasedDeduplicator.java

```diff
@@ -90,7 +86,23 @@ public void deleteKeys(String messageId) {
         (beetleAmqpConfig.getBeetleRedisStatusKeyExpiryIntervalSeconds() > 0)
             ? Arrays.stream(keySuffixes).filter(s -> !s.equals(STATUS))
             : Arrays.stream(keySuffixes);
-    store.delete(suffixStream.map(s -> key(messageId, s)).toArray(String[]::new));
+    store.deleteMultiple(suffixStream.map(s -> key(messageId, s)).toArray(String[]::new));
+  }
+
+  /**
+   * returns true if the initialization succeeds, i.e. there were no already existing STATUS and
+   * EXPIRES keys. return false if at least one of STATUS and EXPIRES keys already exists.
+   */
+  @Override
+  public boolean initKeys(String messageId) {
+    Map<String, Value> keysValues = new HashMap<>();
+    keysValues.put(key(messageId, STATUS), new Value("incomplete"));
+    keysValues.put(
+        key(messageId, EXPIRES),
+        new Value(
+                Instant.now().getEpochSecond()
+                + this.beetleAmqpConfig.getMessageLifetimeSeconds()));
```

Comment:
This is still wrong. Each message can have an individual TTL. You need to use the one that is received with the message.

Reply:
Yes, I was wrong. I will fix it now.